pkg/controller/clusterapi.go

```diff
@@ -41,3 +42,22 @@ func CapiClusterObjectKey(cluster *anywherev1.Cluster) client.ObjectKey {
 		Namespace: constants.EksaSystemNamespace,
 	}
 }
+
+// GetKubeadmControlPlane reads a cluster-api KubeadmControlPlane for an eks-a cluster using a kube client
+// If the KubeadmControlPlane is not found, the method returns (nil, nil).
+func GetKubeadmControlPlane(ctx context.Context, client client.Client, cluster *anywherev1.Cluster) (*controlplanev1.KubeadmControlPlane, error) {
```

Comment:
This is not needed right?

Reply:
I was going to use this in r.DetectUpgrade we use to pull back the kubeadmControlPlane. Is there a better place I should put this logic?

Reply:
I c, this one tries to get the existing CAPI.